Unit test TestDeploy has strong dependency on runner's network environment and rare validation

**Describe the bug**
<!--
A clear and concise description of what the bug is.
If you believe this bug is a security issue, please don't use this template and follow our [security guidelines](/SECURITY.md)
-->
```
--- FAIL: TestDeploy (3.57s)
    deploy_helper_test.go:136:
                Error Trace:    /root/antrea/pkg/antctl/raw/multicluster/deploy/deploy_helper_test.go:136
                Error:          "Get \"https://localhost/api\": dial tcp 127.0.0.1:443: connect: no route to host" does not contain "refused"
                Test:           TestDeploy
FAIL
```
The test always failed in my development environment because it asserts accessing localhost 443 must return "refused" message. This is not something should be asserted in unit test.

Generally, unit tests shouldn't have external dependency on system and network. And this test doesn't really help prevent the tested function `deploy` being broken as there is no useful validation. The only check is to ensure its attempt to deploy will fail with a specific message.

The unit test should either fake the clientset and check whether expected calls are made, or should not be there if no useful validation can be added.

--- a/pkg/antctl/raw/multicluster/deploy/deploy_helper.go
+++ b/pkg/antctl/raw/multicluster/deploy/deploy_helper.go
@@ -51,6 +51,8 @@ const (
 	memberYAML           = "antrea-multicluster-member.yml"
 )
 
+var httpGet = http.Get
+
 func generateManifests(role string, version string) ([]string, error) {
 	var manifests []string
 	switch role {
@@ -162,7 +164,7 @@ func deploy(cmd *cobra.Command, role string, version string, namespace string, f
 		}
 		for _, manifest := range manifests {
 			// #nosec G107
-			resp, err := http.Get(manifest)
+			resp, err := httpGet(manifest)
 			if err != nil {
 				return err
 			}

--- a/pkg/antctl/raw/multicluster/deploy/deploy_helper_test.go
+++ b/pkg/antctl/raw/multicluster/deploy/deploy_helper_test.go
@@ -15,10 +15,14 @@
 package deploy
 
 import (
+	"errors"
+	"io"
 	"log"
+	"net/http"
 	"os"
 	"path/filepath"
 	"reflect"
+	"strings"
 	"testing"
 
 	"github.com/spf13/cobra"
@@ -131,10 +135,38 @@ kind: Config`)
 	kubeconfig := ""
 	cmd := &cobra.Command{}
 	cmd.Flags().StringVarP(&kubeconfig, "kubeconfig", "k", fakeKubeconfig.Name(), "path of kubeconfig")
-	err = deploy(cmd, "leader", "latest", "kube-system", "")
-	if err != nil {
-		assert.Contains(t, err.Error(), "refused")
-	} else {
-		t.Error("Expected to get error but nil")
+	tests := []struct {
+		name        string
+		body        string
+		err         error
+		expectedErr string
+	}{
+		{
+			name:        "error when getting manifests",
+			err:         errors.New("wrong link"),
+			expectedErr: "wrong link",
+		}, {
+			name:        "error with wrong manifests",
+			body:        "fakebody",
+			expectedErr: "json: cannot unmarshal string into Go value",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			httpGet = func(url string) (resp *http.Response, err error) {
+				return &http.Response{Body: io.NopCloser(strings.NewReader(tt.body))}, tt.err
+			}
+			defer func() {
+				httpGet = http.Get
+			}()
+			gotErr := deploy(cmd, "leader", "latest", "kube-system", "")
+			if tt.expectedErr != "" {
+				if gotErr != nil {
+					assert.Contains(t, gotErr.Error(), tt.expectedErr)
+				} else {
+					t.Error("Expected to get error but got nil")
+				}
+			}
+		})
 	}
 }

--- a/pkg/antctl/raw/multicluster/deploy/leader_cluster_test.go
+++ b/pkg/antctl/raw/multicluster/deploy/leader_cluster_test.go
@@ -1,0 +1,69 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deploy
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLeaderValidateAndComplete(t *testing.T) {
+	tests := []struct {
+		name            string
+		namespace       string
+		version         string
+		expectedVersion string
+		expectedErr     string
+	}{
+		{
+			name:        "empty Namesapace",
+			expectedErr: "Namespace must be specified",
+		},
+		{
+			name:            "set version",
+			namespace:       "default",
+			version:         "1.10",
+			expectedVersion: "1.10",
+		},
+		{
+			name:            "not set version",
+			namespace:       "default",
+			expectedVersion: "latest",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			opts := &leaderClusterOptions{
+				namespace:     tt.namespace,
+				antreaVersion: tt.version,
+			}
+			gotErr := opts.validateAndComplete()
+			if gotErr != nil {
+				assert.Equal(t, tt.expectedErr, gotErr.Error())
+			} else {
+				assert.Equal(t, tt.expectedVersion, opts.antreaVersion)
+			}
+		})
+	}
+}
+
+func TestNewLeaderClusterCmd(t *testing.T) {
+	cmd := NewLeaderClusterCmd()
+	assert.Equal(t, cmd.Use, "leadercluster")
+	assert.NotNil(t, cmd.Flags().Lookup("namespace"))
+	assert.NotNil(t, cmd.Flags().Lookup("antrea-version"))
+	assert.NotNil(t, cmd.Flags().Lookup("manifest-file"))
+}

--- a/pkg/antctl/raw/multicluster/deploy/member_cluster_test.go
+++ b/pkg/antctl/raw/multicluster/deploy/member_cluster_test.go
@@ -1,0 +1,69 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deploy
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMemberValidateAndComplete(t *testing.T) {
+	tests := []struct {
+		name            string
+		namespace       string
+		version         string
+		expectedVersion string
+		expectedErr     string
+	}{
+		{
+			name:        "empty Namesapace",
+			expectedErr: "Namespace must be specified",
+		},
+		{
+			name:            "set version",
+			namespace:       "default",
+			version:         "1.10",
+			expectedVersion: "1.10",
+		},
+		{
+			name:            "not set version",
+			namespace:       "default",
+			expectedVersion: "latest",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			opts := &leaderClusterOptions{
+				namespace:     tt.namespace,
+				antreaVersion: tt.version,
+			}
+			gotErr := opts.validateAndComplete()
+			if gotErr != nil {
+				assert.Equal(t, tt.expectedErr, gotErr.Error())
+			} else {
+				assert.Equal(t, tt.expectedVersion, opts.antreaVersion)
+			}
+		})
+	}
+}
+
+func TestNewMemberClusterCmd(t *testing.T) {
+	cmd := NewMemberClusterCmd()
+	assert.Equal(t, cmd.Use, "membercluster")
+	assert.NotNil(t, cmd.Flags().Lookup("namespace"))
+	assert.NotNil(t, cmd.Flags().Lookup("antrea-version"))
+	assert.NotNil(t, cmd.Flags().Lookup("manifest-file"))
+}

